Microsoft.Toolkit.Uwp.UI.Controls/MosaicControl/MosaicControl.cs

```diff
@@ -0,0 +1,1105 @@
+
+        // Using a DependencyProperty as the backing store for IsAnimated.  This enables animation, styling, binding, etc...
+        public static readonly DependencyProperty IsAnimatedProperty =
+            DependencyProperty.Register(nameof(IsAnimated), typeof(bool), typeof(MosaicControl), new PropertyMetadata(false, OnIsAnimatedChange));
+
+        // Using a DependencyProperty as the backing store for AnimationStepX.  This enables animation, styling, binding, etc...
+        public static readonly DependencyProperty AnimationStepXProperty =
+            DependencyProperty.Register(nameof(AnimationStepX), typeof(double), typeof(MosaicControl), new PropertyMetadata(1.0));
+
+        // Using a DependencyProperty as the backing store for AnimationStepY.  This enables animation, styling, binding, etc...
+        public static readonly DependencyProperty AnimationStepYProperty =
+            DependencyProperty.Register(nameof(AnimationStepY), typeof(double), typeof(MosaicControl), new PropertyMetadata(1.0));
+
+        // Using a DependencyProperty as the backing store for AnimationDuration.  This enables animation, styling, binding, etc...
+        public static readonly DependencyProperty AnimationDurationProperty =
+            DependencyProperty.Register(nameof(AnimationDuration), typeof(double), typeof(MosaicControl), new PropertyMetadata(30.0, OnAnimationDuration));
+
+        private FrameworkElement rootElement = null;
+        private Canvas containerElement = null;
+        private TranslateTransform containerTranslate = null;
+        private ImageBrush brushXaml = null;
+
+        private ContainerVisual containerVisual = null;
+        private CompositionSurfaceBrush brushVisual = null;
+
+        private Size imageSize = Size.Empty;
+
+        private UriSurface uriSurface = null;
+        private Visual rootVisual = null;
+
+        private DispatcherTimer timerAnimation = null;
+
+        /// <summary>
+        /// A Scrollviewer used for synchronized the move of the <see cref="MosaicControl"/>
+        /// </summary>
+        private ScrollViewer scrollviewer = null;
+
+        /// <summary>
+        /// a flag to lock shared method
+        /// </summary>
+        private SemaphoreSlim flag = new SemaphoreSlim(1);
+
+        private List<SpriteVisual> compositionChildren = new List<SpriteVisual>(50);
+        private List<Rectangle> xamlChildren = new List<Rectangle>(50);
+
+        private bool isImageSourceLoaded = false;
+        private bool isRootElementSizeChanged = false;
+
+        private CompositionPropertySet propertyOffsetModulo = null;
+        private object lockerOffset = new object();
+
+        private double animationX = 0;
+        private double animationY = 0;
+
+        private enum UIStrategy
+        {
+            /// <summary>
+            /// MosaicControl is created with XAML
+            /// </summary>
+            PureXaml,
```

Comment:
The rendering with XAML has been created at first for the XAML Viewer (when it'll be functionnal ^^) and after for version that doesn't support Composition. As it use a unique ImageBrush it is not too expensive for the memory consumption and the performances are not bad.